File: java/110.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    boolean flag = true;
    public boolean isBalanced(TreeNode root) {
        if(root == null){
            return true;
        }
        depth(root);
        return flag;
        
    }
    public int depth(TreeNode root){
        if(root == null){
            return 0;
        }
        int l = depth(root.left);
        int r = depth(root.right);
        if(l - r > 1 || r - l > 1){
            flag = false;
        }
        return Math.max(l,r) + 1;
    }
}

File: go/110.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
var flag = true

func isBalanced(root *TreeNode) bool {
	if root == nil {
		return true
	}
	_, b := helper(root)
	return b
}

func helper(node *TreeNode) (int, bool) {
	if node == nil {
		return 0, true
	}
	left, b1 := helper(node.Left)
	right, b2 := helper(node.Right)
	b := b1 && b2
	if left-right > 1 || right-left > 1 {
		b = false
	}
	return max(left, right) + 1, b
}

func max(i, j int) int {
	if i > j {
		return i
	}
	return j
}